ibis/pandas/execution/window.py

```diff
@@ -130,24 +132,64 @@ def get_aggcontext_window(
     return aggcontext
 
 
+def trim_with_timecontext(data, timecontext):
+    try:
+        begin, end = map(pd.to_datetime, timecontext)
```

Comment:
Let's not convert types here. timecontext should already be [pd.Timestamp, pd.Timestamp]

Reply:
Well,it is not guaranteed to be ``pd.Timestamp`` if this op is at root level, also it's not guaranteed if none of its ancestor nodes implement a ``compute_time_context``, ( the default behavior is just passing whatever ``timecontext`` is without converting them) Maybe we should add another layer to convert first if checking here seems strange?

Reply:
Add check in main_execute for time context, will raise an exception before any execution